### PR TITLE
Fix flaky test by avoiding refused connection with short timeout

### PR DESCRIPTION
**Pull request by pquentin**
As seen in https://github.com/urllib3/urllib3/runs/1193400148

### CODE REVIEW

File: test/with_dummyserver/test_socketlevel.py [hunk 1]
```diff
@@ -458,7 +458,7 @@ def socket_handler(listener):
 
         self._start_server(socket_handler)
         with HTTPConnectionPool(
-            self.host, self.port, timeout=SHORT_TIMEOUT, retries=True
+            self.host, self.port, timeout=LONG_TIMEOUT, retries=True
```
Comment (sethmlarson) [inline test/with_dummyserver/test_socketlevel.py:461]: Can we rename this test to match new retry option terminology?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
One comment then LGTM

File: test/with_dummyserver/test_socketlevel.py (hunk 1 above)
Comment (pquentin) [reply test/with_dummyserver/test_socketlevel.py:461]: Done, thanks

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**